### PR TITLE
Add an helpful textbox to warn players in Adult shooting gallery

### PR DESCRIPTION
**Pull request by GSKirox**
This is basically a copy of what is done is OoT3D Rando, so shoutouts to them !
If the player wins Adult shooting gallery without having a bow, they'll get 50 rupees but not the actual item. Vanilla behaviour, but definitely something that can trip up new players.

This adds an helpful textbox in that case that says "Come back when you have your own bow and you'll get the real prize!".

This textbox will fade out automatically after 120 frames, or will disappear faster if you play the game again/read the sign/pull out the ocarina/open the door, so there's a chance a new player will still miss it before reading the message, but at least we tried !
Also it will only show once by save/quit, to try to not annoy players that actually know about this and wants to farm the 50 rupees.

This textbox is explicitely excluded from text shuffle, for obvious reasons.

### CODE REVIEW

**Comment by GSKirox** [comment]
Here is what this looks like in game


https://github.com/OoTRandomizer/OoT-Randomizer/assets/65768236/8adac179-4cb3-44b1-b662-5fecf59a03fa

File: ASM/c/message.c [hunk 1]
```diff
@@ -121,3 +122,32 @@ bool Message_Decode_Additional_Control_Codes(uint8_t currChar, uint32_t* pDecode
...
+uint8_t shooting_gallery_show_message = 0;
+// Displays a warning message if player did adult shooting gallery without bow.
+void shooting_gallery_message() {
+    // Child/Adult shooting galleries actor is the same, so the asm hook will work for both.
+    // We only want the message for Adult.
+    if (!LINK_IS_ADULT) {
+        return;
+    }
+    // Check if we have a bow.
+    if (!(z64_file.items[ITEM_BOW] == -1)) {
```
Comment (fenhl) [inline ASM/c/message.c]:
```suggestion
    if (z64_file.items[ITEM_BOW] != -1) {
```

**Review by fenhl — CHANGES_REQUESTED** [review]
Small code style nit, looks good other than that

**Comment by GSKirox** [comment]
Rebased with requested change done.

**Review by fenhl — APPROVED** [review]

**Pull request merged**